src/compiler/factory/node_factory.rs

```diff
@@ -298,7 +336,24 @@ impl NodeFactory {
         elements: TElements, /*Expression*/
     ) -> ArrayLiteralExpression {
         let node = self.create_base_expression(base_factory, SyntaxKind::ArrayLiteralExpression);
-        let elements_array = self.create_node_array(elements);
+        let elements_as_node_array = match elements.into() {
+            NodeArrayOrVec::NodeArray(node_array) => node_array,
+            NodeArrayOrVec::Vec(elements) => NodeArray::new(elements),
+        };
+        let has_trailing_comma = {
+            let last_element =
+                last_or_undefined(Into::<&[Rc<Node>]>::into(&elements_as_node_array));
+            if let Some(last_element) = last_element {
+                if is_omitted_expression(&**last_element) {
+                    Some(true)
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
```

Comment:
I'd write this as an `and_then` on `last_element`.

Reply:
Sure, updated to use `.and_then()`